src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -91,7 +91,7 @@ private TupleTypeSymbol(ImmutableArray<Location> locations, NamedTypeSymbol unde
 
             NamedTypeSymbol underlyingType = GetTupleUnderlyingType(elementTypes, syntax, compilation, diagnostics);
 
-            if (((SourceModuleSymbol)compilation.SourceModule).AnyReferencedAssembliesAreLinked)
+            if (((SourceModuleSymbol)compilation.SourceModule).AnyReferencedAssembliesAreLinked && diagnostics != null)
```

Comment:
Yes, that would be a correct change for VB. It won't be hit by any scenario though.